components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1353,6 +1378,41 @@ _resolve_dependencies() {
   _validate_deps
 }
 
+# **Internal**  Build and export `$PATH` containing each path in our own
+# `${pkg_bin_dirs[@]}` array, and then any dependency's `PATH` entry (direct or
+# transitive) if one exists. The ordering of the path is specific to
+# `${pkg_all_tdeps_resolved[@]}` which is further explained in the
+# `_resolve_dependencies()` function.
+#
+# Reference implementation:
+# https://github.com/habitat-sh/habitat/blob/3d63753468ace168bbbe4c52e600d408c4981b03/components/plan-build/bin/hab-plan-build.sh#L1584-L1638
+_set_build_path() {
+  local paths=()
+  local dir dep data
+
+  # Add element for each entry in `$pkg_bin_dirs[@]` first
+  for dir in "${pkg_bin_dirs[@]}"; do
+    paths+=("$pkg_prefix/$dir")
+  done
+
+  for dep in "${pkg_all_tdeps_resolved[@]}"; do
+    if [[ -f "$dep/PATH" ]]; then
+      data="$(cat "$dep/PATH")"
+      data="$(trim "$data")"
+      while read -r entry; do
+        paths=($(_return_or_append_to_set "$entry" "${paths[@]}"))
```

Comment:
Same comment as above, re: filtering `PATH` metadata file entries from packages other than the current one.

Reply:
Same as above, will add this in!